netCDF error in unit tests

I stumbled over this error on x86_64 and i686:
```
[ RUN      ] OptionsNetCDFTest.FieldPerpWriteCellCentre
unknown file: Failure
C++ exception with description "NetCDF: HDF error
file: ../../cxx4/ncFile.cpp  line:88" thrown in the test body.
[  FAILED  ] OptionsNetCDFTest.FieldPerpWriteCellCentre (559 ms)
```

And this one on s390x:
```
----- test-io -----
Making I/O test
Reading benchmark data
Traceback (most recent call last):
  File "/builddir/build/BUILD/BOUT++-v4.4.0/build_mpich/tests/integrated/test-io/./runtest", line 42, in <module>
    bmk[v] = collect(v, path="data", prefix="benchmark.out", info=False)
  File "/usr/lib/python3.10/site-packages/boutdata/collect.py", line 285, in collect
    f = getDataFile(0)
  File "/usr/lib/python3.10/site-packages/boutdata/collect.py", line 189, in getDataFile
    return DataFile(file_list[i])
  File "/usr/lib/python3.10/site-packages/boututils/datafile.py", line 102, in __init__
    self.impl = DataFile_netCDF(
  File "/usr/lib/python3.10/site-packages/boututils/datafile.py", line 372, in __init__
    self.open(filename, write=write, create=create, format=format)
  File "/usr/lib/python3.10/site-packages/boututils/datafile.py", line 346, in open
    self.handle = Dataset(filename, "r")
  File "src/netCDF4/_netCDF4.pyx", line 2353, in netCDF4._netCDF4.Dataset.__init__
  File "src/netCDF4/_netCDF4.pyx", line 1926, in netCDF4._netCDF4._get_vars
AttributeError: 'NoneType' object has no attribute 'dimensions'
======= 1 failed in 617.01 seconds ========
make: *** [makefile:49: check-integrated-tests] Error 1
```

The error seems to (at least on x86_64) non deterministic. I haven't observed it locally.

It might be a bug in hdf5 :-(

--- a/tests/unit/sys/test_options_netcdf.cxx
+++ b/tests/unit/sys/test_options_netcdf.cxx
@@ -224,6 +224,10 @@ TEST_F(OptionsNetCDFTest, FieldPerpWriteCellCentre) {
     fperp.setIndex(2);
     options["fperp"] = fperp;
 
+    // Ensure MPI is initialised, otherwise we end up creating threads while
+    // the file is open, and the lock is not removed on closing.
+    fperp.getMesh->getXcomm();
+
     // Write file
     OptionsNetCDF(filename).write(options);
   }

--- a/tests/unit/sys/test_options_netcdf.cxx
+++ b/tests/unit/sys/test_options_netcdf.cxx
@@ -226,7 +226,7 @@ TEST_F(OptionsNetCDFTest, FieldPerpWriteCellCentre) {
 
     // Ensure MPI is initialised, otherwise we end up creating threads while
     // the file is open, and the lock is not removed on closing.
-    fperp.getMesh->getXcomm();
+    fperp.getMesh()->getXcomm();
 
     // Write file
     OptionsNetCDF(filename).write(options);

